src/book.py

```diff
@@ -176,6 +178,79 @@ def _read_coinbase(self, file_path: Path) -> None:
                     self.append_operation("Fee", utc_time, platform,
                                           eur_fee, "EUR", row, file_path)
 
+    def _read_kraken_trades(self, file_path: Path) -> None:
+        log.error(f"{file_path.name}: "
+                  f"Looks like this is a Kraken 'Trades' history, but we need the 'Ledgers' history. "
+                  f"(See: https://github.com/provinzio/CoinTaxman/wiki/Exchange:-Kraken)")
+
+    def _read_kraken_ledgers(self, file_path: Path) -> None:
+        platform = "kraken"
+        operation_mapping = {
+            "trade": None,  # Buy or Sell order
+            "spend": "Sell",  # Sell ordered via 'Buy Crypto' button
+            "receive": "Buy",  # Buy ordered via 'Buy Crypto' button
+            "transfer": "Airdrop",
+            "reward": "StakingInterest",
+            "deposit": "Deposit",
+            "withdrawal": "Withdraw",
+        }
+
+        # Need to track state of "duplicate entries" for deposits / withdrawals:
+        #   dup_state["deposit"] == 1: Deposit is broadcast to blockchain            <-- Taxable event (is in public trade history)
+        #   dup_state["deposit"] == 2: Deposit is credited to Kraken account         <-- Skipped
+        #   dup_state["withdrawal"] == 1: Withdrawal is requested in Kraken account  <-- Skipped
+        #   dup_state["withdrawal"] == 2: Withdrawal is broadcast to blockchain      <-- Taxable event (is in public trade history)
+        dup_state, dup_skip = {"deposit": 0, "withdrawal": 0}, {"deposit": 2, "withdrawal": 1}
+        # See: https://support.kraken.com/hc/en-us/articles/360001169443-Why-there-are-duplicate-entries-for-deposits-withdrawals
+
+        with open(file_path, encoding="utf8") as f:
+            reader = csv.reader(f)
+
+            # Skip header.
+            next(reader)
+
+            for txid, refid, _utc_time, _type, subtype, aclass, _asset, _amount, _fee, balance in reader:
+                row = reader.line_num
+
+                # Skip "duplicate entries" for deposits / withdrawals
+                if _type in dup_state.keys():
+                    dup_state[_type] = (dup_state[_type] + 1) % 2
+                    if dup_state[_type] == (dup_skip[_type] + 1) % 2:
+                        continue
+
+                # Parse data.
+                utc_time = datetime.datetime.strptime(_utc_time, "%Y-%m-%d %H:%M:%S")
+                utc_time = utc_time.replace(tzinfo=tzutc())
+                change = float(_amount)
+                coin = kraken_fiat_map.get(_asset, kraken_crypto_map.get(_asset, _asset))
```

Comment:
While in principle we could, I'd personally find that inelegant, since `kraken_fiat_map` maps a single asset, while `kraken_crypto_map` maps an asset-_pair_.

Reply:
I do not understand you. `kraken_fiat_map` maps fiat, `kraken_crypto_map` maps crypto. We could merge them. Both map the internal Kraken asset to the real asset. I would not change `kraken_pair_map`.

Reply:
Sorry, I confused `kraken_crypto_map` with `kraken_pair_map`. Yes, I'll merge them.